python/packages/core/agent_framework/_workflows/_typing_utils.py

```diff
@@ -4,56 +4,72 @@
 from collections.abc import Mapping
 from dataclasses import fields, is_dataclass
 from types import UnionType
-from typing import Any, Union, get_args, get_origin
+from typing import Any, TypeVar, Union, cast, get_args, get_origin
 
 logger = logging.getLogger(__name__)
 
+T = TypeVar("T")
```

Comment:
Note: the changes in this file improve typing and remove the current Pylance errors that are present.